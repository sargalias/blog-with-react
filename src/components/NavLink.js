import React from 'react';

const NavLink = ({link, label, isActive=false}) => (
  <li className={`nav-item ${isActive && 'active'}`}>
    <a
      className="nav-link"
      href={link}
    >
      {label} {isActive && <span className="sr-only">(current)</span>}
    </a>
  </li>
);

export default NavLink;
