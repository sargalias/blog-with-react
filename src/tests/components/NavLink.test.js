import React from 'react';
import {shallow} from 'enzyme';
import NavLink from '../../components/NavLink';

test('NavLink should render correctly with no props', () => {
  const wrapper = shallow(<NavLink />);
  expect(wrapper).toMatchSnapshot();
});

test('NavLink should render correctly with props and isActive false', () => {
  const wrapper = shallow(<NavLink link='/test' label='test' isActive={false} />);
  expect(wrapper).toMatchSnapshot();
});

test('NavLink should render correctly with props and isActive false', () => {
  const wrapper = shallow(<NavLink link='/test' label='test' isActive={true} />);
  expect(wrapper).toMatchSnapshot();
});

test('NavLink has correct label', () => {
  const label = 'test text';
  const re = new RegExp(label);
  const wrapper = shallow(<NavLink label={label} />);
  expect(wrapper.text()).toMatch(re);
});

test('NavLink is set correctly when isActive=true', () => {
  const isActive = true;
  const expected = ' (current)';
  const wrapper = shallow(<NavLink isActive={isActive} />);
  expect(wrapper.text()).toBe(expected);
});

test('NavLink is set correctly when isActive=false', () => {
  const isActive = false;
  const expected = ' ';
  const wrapper = shallow(<NavLink isActive={isActive} />);
  expect(wrapper.text()).toBe(expected);
});
