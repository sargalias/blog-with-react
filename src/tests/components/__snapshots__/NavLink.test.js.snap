// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NavLink should render correctly with no props 1`] = `
<li
  className="nav-item false"
>
  <a
    className="nav-link"
  >
     
  </a>
</li>
`;

exports[`NavLink should render correctly with props and isActive false 1`] = `
<li
  className="nav-item false"
>
  <a
    className="nav-link"
    href="/test"
  >
    test
     
  </a>
</li>
`;

exports[`NavLink should render correctly with props and isActive false 2`] = `
<li
  className="nav-item active"
>
  <a
    className="nav-link"
    href="/test"
  >
    test
     
    <span
      className="sr-only"
    >
      (current)
    </span>
  </a>
</li>
`;
